src/typescript-service.ts

```diff
@@ -338,6 +338,847 @@ export default class TypeScriptService {
 		});
 	}
 
+	getDocumentSymbol(params: DocumentSymbolParams): Promise<SymbolInformation[]> {
+		const uri = params.textDocument.uri;
+		return this.ensureFilesForHoverAndDefinition(uri).then(() => {
+			const fileName = util.uri2path(uri);
+			const config = this.projectManager.getConfiguration(uri);
+			return config.prepare(this.connection).then(() => {
+				this.projectManager.syncConfiguration(config, this.connection);
```

Comment:
> If I omit syncConfiguration, I get no results back. To test this, remove this line and run npm test.

Addressed in @327a735e16da2782439d717160817f966bb1ffdb
You should use our `TypescriptService::getSourceFile` instead of `Program::getSourceFile`, this way we'll put missing files to TS program and there will be no need to sync the whole configuration